core/renderers/common/constants.js

```diff
@@ -508,9 +502,13 @@ Blockly.blockRendering.ConstantProvider = function() {
 
 /**
  * Initialize shape objects based on the constants set in the constructor.
+ * @param {!Blockly.Theme} theme The workspace theme object.
  * @package
  */
-Blockly.blockRendering.ConstantProvider.prototype.init = function() {
+Blockly.blockRendering.ConstantProvider.prototype.init = function(theme) {
```

Comment:
I think this needs to be flagged for doc updates as well. Subclasses need to pass the theme up to this parent, and I think currently they will not.

Reply:
This is why I was asking about this change--it requires anyone who has already started tinkering with it to update their constructor code, and I'm not sure we really gain anything over calling setTheme after init.

Reply:
Thanks for pointing that out, I like that we're being diligent about unnecessary breaking changes.
I actually think setTheme needs to be called before init in case any dynamic constants are relied upon in any of the init methods, eg: makeNotch, but I agree we don't gain much, will refactor to call setTheme before init.

Reply:
Done.